src/logic/community/broadcaster.ts

```diff
@@ -154,51 +187,107 @@ export function createCommunityBroadcasterComponent(
+   * @param {BroadcastableEvent} event - The event to broadcast
+   * @throws {Error} If excluded address or community ID is not found in metadata
+   */
+  async function broadcastToAllMembersButExcluded(event: BroadcastableEvent): Promise<void> {
+    const {
+      metadata: { authorAddress, communityId }
+    } = event as CommunityPostAddedEvent
+
+    const excludedAddress = authorAddress.toLowerCase()
+
+    const allMemberAddresses = await getAllCommunityMembersAddresses(communityId)
+    const addressesToNotify = allMemberAddresses.filter((address) => address.toLowerCase() !== excludedAddress)
+    const memberBatches = createMemberBatches(addressesToNotify)
+
+    await sns.publishMessages(
+      memberBatches.map((batch, i) => ({
         ...event,
-        subType: Events.SubType.Community.DELETED,
+        key: `${event.key}-batch-${i + 1}`,
         metadata: {
-          id: event.metadata.id,
-          name: event.metadata.name,
-          thumbnailUrl: event.metadata.thumbnailUrl
+          ...event.metadata,
+          addressesToNotify: batch
         }
-      } as CommunityDeletedEventReducedMetadata
-      await publishToAllMembersButOwner(communityDeletedEvent)
-    } else {
-      await sns.publishMessage(event)
+      }))
+    )
+  }
+
+  /**
+   * Broadcasts deleted content violation - notify owner with original event,
+   * then notify other members with a DELETED event
+   * @param {BroadcastableEvent} event - The content violation event to broadcast
+   */
+  async function broadcastDeletedContentViolation(event: BroadcastableEvent): Promise<void> {
+    const violationEvent = event as CommunityDeletedContentViolationEvent
+
+    // Notify the owner with the original content violation event
+    await sns.publishMessage(violationEvent)
+
+    // Send a different notification to the rest of the community members
+    const communityDeletedEvent: CommunityDeletedEventReducedMetadata = {
+      ...violationEvent,
+      subType: Events.SubType.Community.DELETED,
+      metadata: {
+        id: violationEvent.metadata.id,
+        name: violationEvent.metadata.name,
+        thumbnailUrl: violationEvent.metadata.thumbnailUrl
+      }
     }
+
+    await broadcastToAllMembersButOwner(communityDeletedEvent)
+  }
+
+  /**
+   * Direct broadcast - publish event as-is without any modifications
+   * @param {BroadcastableEvent} event - The event to broadcast
+   */
+  async function directBroadcast(event: BroadcastableEvent): Promise<void> {
+    await sns.publishMessage(event)
+  }
+
+  /**
+   * Creates the broadcasting registry with all strategies
+   * @returns {BroadcastingRegistry} Map of event subTypes to their broadcasting handlers
+   */
+  function createBroadcastingRegistry(): BroadcastingRegistry {
+    const registry = new Map<Events.SubType.Community, BroadcastingEventHandler>()
+
+    // Events that should be broadcasted to all members
+    registry.set(Events.SubType.Community.DELETED, broadcastToAllMembers)
```

Comment:
Since the Owner is the only role that could delete the Community, could we double-check with product if they should receive this `DELETED` event?

Mostly because if we can avoid delivering it to them, we could handle the `DELETED_CONTENT_VIOLATION` differently and maybe more meaningful, here is my idea:
- Whenever `DELETED_CONTENT_VIOLATION` happens, we could trigger two events:
  - `COMMUNITY_DELETED`: to be notified to all members but owners (if Product agree with us)
  -  `DELETED_CONTENT_VIOLATION`: to be notified to only owners

This way we could set:
```
registry.set(Events.SubType.Community.DELETED, broadcastToAllMembersButOwners)
registry.set(Events.SubType.Community.DELETED_CONTENT_VIOLANTION, broadcastToOwnersAndModerators) // or broadcastToOwner
```

Reply:
I really like this idea